recipes/sqlcipher/all/conanfile.py

```diff
@@ -0,0 +1,177 @@
+from conans import tools, ConanFile, AutoToolsBuildEnvironment, RunEnvironment
+import platform
+import sys
+import os
+
+
+class SqlcipherConan(ConanFile):
+    name = "sqlcipher"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.zetetic.net/sqlcipher/"
+    description = "SQLite extension that provides 256 bit AES encryption of database files."
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False], "crypto_library": ["openssl", "libressl"]}
+    default_options = {"shared": False, "fPIC": True, "crypto_library": "openssl"}
+    topics = ("database", "encryption", "SQLite")
+    exports_sources = "patches/*"
+    generators = "cmake"
+    _source_subfolder = "source_subfolder"
+
+
+    def config_options(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def build_requirements(self):
+        self.build_requires("tcl/8.6.10")
+
+    def requirements(self):
+        if self.options.crypto_library == "openssl":
+            self.requires("openssl/1.1.1d")
+        else:
+            self.requires("libressl/2.9.2")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _build_visual(self):
+        crypto_dep = self.deps_cpp_info[str(self.options.crypto_library)]
+        crypto_incdir = crypto_dep.include_paths[0]
+        crypto_libdir = crypto_dep.lib_paths[0]
+        libs = map(lambda lib : lib + ".lib", crypto_dep.libs)
+
+        nmake_flags = [
+                "TLIBS=\"%s\"" % " ".join(libs),
+                "LTLIBPATHS=/LIBPATH:%s" % crypto_libdir,
+                "OPTS=\"-I%s -DSQLITE_HAS_CODEC\"" % (crypto_incdir),
+                "NO_TCL=1",
+                "USE_AMALGAMATION=1",
+                "OPT_FEATURE_FLAGS=-DSQLCIPHER_CRYPTO_OPENSSL",
+                "TCLSH_CMD=%s" % self.deps_env_info.TCLSH,
+                ]
+
+        main_target = "dll" if self.options.shared else "sqlcipher.lib"
+
+        if self.settings.compiler.runtime in ["MD", "MDd"]:
+            nmake_flags.append("USE_CRT_DLL=1")
+        if self.settings.build_type == "Debug":
+            nmake_flags.append("DEBUG=2")
+        if platform.release() == "10":
```

Comment:
does it mean library built on Windows 10 won't be compatible with lower Windows versions?
also, does it mean resulting library will be different depending if it was built on Windows 10 or another Windows Version?

Reply:
Looking at the `Makefile.msc`:

```
# Set this non-0 to compile binaries suitable for the Windows 10 platform.
#
!IFNDEF FOR_WIN10
FOR_WIN10 = 0
!ENDIF

...

# When compiling for the Windows 10 platform, the PLATFORM macro must be set
# to an appropriate value (e.g. x86, x64, arm, arm64, etc).
#
!IF $(FOR_WIN10)!=0
!IFNDEF PLATFORM
!ERROR Using the FOR_WIN10 option requires a value for PLATFORM.
!ENDIF
!ENDIF
```

I'll try to unconditionally enable `FOR_WIN10`, it seems that it's required in order for built binaries to run everywhere.